module/icp/asm-x86_64/sha2/sha256_avx2.S

```diff
@@ -0,0 +1,788 @@
+########################################################################
```

Comment:
These assembly files were taken from `linux-5.14.1/arch/x86/crypto/`

They are all dual licensed GPL and BSD.

Are there any issues with this?

For a diff of the changes:

```bash
mkdir -p linux
pushd linux

wget -c "https://raw.githubusercontent.com/torvalds/linux/v5.14/arch/x86/crypto/sha256-avx-asm.S"
wget -c "https://raw.githubusercontent.com/torvalds/linux/v5.14/arch/x86/crypto/sha256-avx2-asm.S"
wget -c "https://raw.githubusercontent.com/torvalds/linux/v5.14/arch/x86/crypto/sha256-ssse3-asm.S"
wget -c "https://raw.githubusercontent.com/torvalds/linux/v5.14/arch/x86/crypto/sha256_ni_asm.S"

diff -u sha256-avx-asm.S ../module/icp/asm-x86_64/sha2/sha256_avx.S
diff -u sha256-avx2-asm.S ../module/icp/asm-x86_64/sha2/sha256_avx2.S
diff -u sha256-ssse3-asm.S ../module/icp/asm-x86_64/sha2/sha256_ssse3.S
diff -u sha256_ni_asm.S ../module/icp/asm-x86_64/sha2/sha256_ni.S

popd
```